tools/rosbag/src/rosbag/migration.py

```diff
@@ -169,9 +169,16 @@ def fixbag2(migrator, inbag, outbag, force=False):
         for topic, msg, t, conn_header in bag.read_messages(raw=True, return_connection_header=True):
             new_msg_type = migrator.find_target(msg[4])
             if new_msg_type != None:
-                mig_msg = migrator.migrate_raw(msg, (new_msg_type._type, None, new_msg_type._md5sum, None, new_msg_type))
-                new_conn_header = _migrate_connection_header(conn_header, new_msg_type)
-                rebag.write(topic, mig_msg, t, connection_header=new_conn_header, raw=True)
+                try:
+                    mig_msg = migrator.migrate_raw(msg, (new_msg_type._type, None, new_msg_type._md5sum, None, new_msg_type))
+                    new_conn_header = _migrate_connection_header(conn_header, new_msg_type)
+                    rebag.write(topic, mig_msg, t, connection_header=new_conn_header, raw=True)
+                except BagMigrationException:
+                    if force:
+                        # migration failed, copy original message
+                        rebag.write(topic, msg, t, connection_header=conn_header, raw=True)
+                    else:
+                        raise
```

Comment:
this needs to be caught on here to avoid abort?

https://github.com/ros/ros_comm/blob/b6c23e1282815823c8e07b47384391cf7cf0eab4/tools/rosbag/src/rosbag/rosbag_main.py#L456-L457

Reply:
That makes sense. I also added it there.